### PR TITLE
server-mux: Read timeout only after conn creation

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
*) Multiple setting of read timeout in Read() is disabled, the server
   requires read timeout only after connection establishment.
*) Increase read timeout to 30 seconds
*) Conn Mux was using double buffer which is useless (typo)

## Motivation and Context
Cleanup, probably f.i.x.e.s #4419 

## How Has This Been Tested?
go test + manual

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by harshavardhana — DISMISSED** [review]

File: cmd/server-mux.go [hunk 1]
```diff
@@ -133,11 +132,6 @@ func (c *ConnMux) Read(b []byte) (n int, err error) {
 		return n, err
```
Comment (fwessels) [inline cmd/server-mux.go:129]: Comments in lines 119 through to 122 no longer accurate

**Review by fwessels — APPROVED** [review]
See comment, for the rest LGTM

File: cmd/server-mux.go (hunk 1 above)
Comment (harshavardhana) [inline cmd/server-mux.go]: Cool.

**Comment by balamurugana** [comment]
This PR https://github.com/minio/minio/pull/4217 is already taking care of this.  I strongly recommend to review/merge https://github.com/minio/minio/pull/4217

**Comment by harshavardhana** [comment]
> This PR #4217 is already taking care of this. I strongly recommend to review/merge #4217

Only one part @balamurugana - not everything. Not having deadlines for Read() on net.Conn is still needed for RPC clients. This PR is not a complete fix either.. i just added a comment on your PR and added some reviews as well.

File: cmd/server-mux.go (hunk 1 above)
Comment (vadmeste) [reply cmd/server-mux.go:129]: > Comments in lines 119 through to 122 no longer accurate

Thanks, done.

**Comment by vadmeste** [comment]
@harshavardhana, shall we totally disable reading timeout even for peeking ? Because RPC still uses this code path for communication between servers.

**Comment by harshavardhana** [comment]
Peeking is fine @vadmeste it is quite useful there. We need to address the RPC separately.

**Comment by harshavardhana** [comment]
@vadmeste @balamurugana PR is handling this case in the refactor -  you can review the changes done by @balamurugana - I think we can close this PR and concentrate on #4217

**Comment by vadmeste** [comment]
> @vadmeste @balamurugana PR is handling this case in the refactor - you can review the changes done by @balamurugana - I think we can close this PR and concentrate on #4217

Agree, it will be easier.

**Comment by dcharbonnier** [comment]
:+1:  upgrading all hosts to the edge now, let's see, arf ok not merged ;-)

**Comment by vadmeste** [comment]
> upgrading all hosts to the edge now, let's see, arf ok not merged ;-)

@dcharbonnier we are currently reviewing this PR, https://github.com/minio/minio/pull/4217, I already tested it, it looks fine. hope we will merge it soon.

**Pull request closed without merging**